### PR TITLE
Decorate GtkEntry fields with CSS to ensure that they are visible

### PR DESCRIPTION
**Pull request by tpimh**
The code is based on [this patch](https://github.com/solvespace/solvespace/issues/184#issuecomment-277509013), I only changed CSS code.

Before:
before
After:
after

### CODE REVIEW

File: src/platform/gtkmain.cpp [hunk 1]
```diff
@@ -1410,6 +1411,19 @@ int main(int argc, char** argv) {
 
     Gtk::Main main(argc, argv);
 
+    // Add our application-specific styles, to override GTK defaults.
+    Glib::RefPtr<Gtk::CssProvider> style_provider = Gtk::CssProvider::create();
+    style_provider->load_from_data(R"(
+      entry {
+        background: black;
+        color: white;
+        font-size: 12px;
```
Comment (whitequark) [inline src/platform/gtkmain.cpp]: This isn't either correct or necessary, the font size is set explicitly.

**Comment by whitequark** [comment]
I've applied a different CSS in master, thanks for discovering the necessary trick for this.

**Comment by tpimh** [comment]
Now it looks like this:
new
I personally would prefer white text on black background, because the interface scheme is dark and dark textbox fits better. That's just a matter of taste.
I agree, overriding font-size here is not a good idea. I think that 16px font is too large and 12px would be optimal. Does it look different with different DPI or GTK+ theme?
Does `font_height` of `16` look oversized on Win32 and Mac?

**Comment by whitequark** [comment]
> I agree, overriding font-size here is not a good idea. I think that 16px font is too large and 12px would be optimal. Does it look different with different DPI or GTK+ theme?

Looks like this for me:
screenshot_20170319_195157

And, since the font size is overridden and inferred from text window font size, the theme should not matter.

By the way, please don't send pull requests, I cannot accept them as there is no CLA yet.

**Comment by whitequark** [comment]
> Does font_height of 16 look oversized on Win32 and Mac?

Oh, and this is irrelevant because we don't use GTK on any platform except Linux.

**Comment by tpimh** [comment]
`CHAR_HEIGHT` is a constant and it is 16 for all platforms.
On your setup textbox looks well-centered vertically and doesn't seem oversized. Probably it's Adwaita's problem.

**Comment by whitequark** [comment]
> CHAR_HEIGHT is a constant and it is 16 for all platforms.

This doesn't matter. The size of the text entry in the graphics window varies, as it fits the constraint label.

> Probably it's Adwaita's problem.

No. The theme should not matter.

**Pull request closed without merging**